From ca6c9b49b868c971021e0299fc370b7b472abc2f Mon Sep 17 00:00:00 2001
From: henry shao <hshao@netlogicmicro.com>
Date: Wed, 23 Mar 2011 17:12:25 -0700
Subject: [PATCH 212/565] bcm-xlp: make Makefile, Kconfig change

Add nand-xlp platform device to support xlp_plat_nand driver.

Based on Broadcom SDK 2.3.

Signed-off-by: henry shao <hshao@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/platform.c | 1 +
 drivers/mtd/nand/Makefile         | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/arch/mips/netlogic/xlp/platform.c b/arch/mips/netlogic/xlp/platform.c
index b4240d6..661ba29 100644
--- a/arch/mips/netlogic/xlp/platform.c
+++ b/arch/mips/netlogic/xlp/platform.c
@@ -137,6 +137,7 @@ struct dev2drv dev2drv_table[MAX_DEV2DRV] = {
 	{XLP_DEVID_MMC,	 "mmc-xlp",	8,	0, 	PLAT_DRV},
 	{XLP_DEVID_SPI,	 "spi-xlp",	8,	0, 	PLAT_DRV},
 	{XLP_DEVID_NOR,	 "nor-xlp",	8,	0, 	PLAT_DRV},
+	{XLP_DEVID_NAND, "nand-xlp",	9,	0, 	PLAT_DRV},
 	{0x0, 			 "",	0,	0,	PLAT_DRV},
 };
 
diff --git a/drivers/mtd/nand/Makefile b/drivers/mtd/nand/Makefile
index bb81891..f9de1f7 100644
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -1,6 +1,9 @@
 #
 # linux/drivers/nand/Makefile
 #
+ifeq ($(CONFIG_NLM_XLP),y)
+        EXTRA_CFLAGS := $(CFLAGS) -DNLM_HAL_LINUX_KERNEL
+endif
 
 obj-$(CONFIG_MTD_NAND)			+= nand.o
 obj-$(CONFIG_MTD_NAND_ECC)		+= nand_ecc.o
-- 
1.8.4.93.g57e4c17

